app/helpers/loss_report_helper.rb

```diff
@@ -0,0 +1,185 @@
+        # Nokogiri does not directly map to an item that is changed thus it still contains xml - this is the
+        # purpose of hash_map), items that represent xml attribute changes, and normal changes.
+        if is_xml?(element)
+          element = Hash.from_xml(element)
+          hash_map(element).each do |item|
+            arr_path = top_level_arr_path("#{path}/#{item['path']}", orig_h, conv_h)
+            # this layer of if/else structure is used to separately evaluate implicit array changes in the xml.
+            # This is why arr_path will evaluate true if the element in question is an array
+            if arr_path && path_not_checked?("#{path}/#{item['path']}", arr_paths)
+              if path_not_checked?(arr_path, arr_paths)
+                arr_paths << arr_path
+                array_comparison(arr_path, orig_h, conv_h).each { |item| add_to_report(item[0], item[1], item[2], hide_items, display, json_output, text_output) }
+              end
+            elsif path_not_checked?("#{path}/#{item['path']}", arr_paths)
+              add_to_report(change, item['value'], "#{path}/#{item['path']}", hide_items, display, json_output, text_output)
+            end
+          end
+        elsif (attr,val = is_attribute?(element))
+          add_to_report(change, val, "#{path}/#{attr}" , hide_items, display, json_output, text_output)
+        else
+          add_to_report(change, element, path, hide_items, display, json_output, text_output)
+        end
+      end
+    end
+    return text_output if display == 'text'
+    return json_output if display == 'json'
+  end
+
+  def is_xml?(element)
+    # checks if the element being passed is xml
+    # may be beneficial to add more checks
+    element.include?('<' && '</' && '>') 
+  end
+
+  def is_attribute?(element)
+    # this method checks if the element being passed is an attribute change;
+    # TODO: it may be beneficial to add more conditions to improve accuracy
+    if element.include?('=') && !element.include?(' = ')
+      attr_val = Array.new
+      element.split('=').each {|item| attr_val << item.strip.delete('\\"')}
+      attr_val
+    else
+      false
+    end
+  end
+
+  def path_not_checked?(arr_path, arr_paths)
+    # this method checks the arr_paths array to see if the path being added to
+    # the report has already been previously evaluated and added
+    arr_paths.each { |path| return false if arr_path.include?(path) }
+    true
+  end
+
+  def top_level_arr_path(path, orig_h, conv_h)
+    # if an array is passed that passes through an array ie. /Contacts/Contact[0]/Role/Name
+    # this method would return /Contacts/Contact because Contact is the outermost array (or false if the path doesn't contain an array)
+    pre_translation_array, pre_translation_path = hash_navigation(path, orig_h)
+    post_translation_array, post_translation_path = hash_navigation(path, conv_h)
+
+    return false if pre_translation_array == false && post_translation_array == false
```

Comment:
we talked about this, might be worth a comment explaining what scenario this is informing us about